### ISSUE TITLE
Add support for Ubuntu 22.04

### ISSUE THREAD

**Issue by pb8o** [aws/ec2-image-builder-roadmap#81]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?

Currently there are images for Ubuntu LTS 16.04, 18.04, 20.04, but not 22.04

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

**Are you currently working around this issue?**
How are you currently solving this problem?

We aren't currently working around it, but it would be nice to have these for running in our CI

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by rswail** [comment]
This is an obvious need. 22.04 is the current LTS release of Ubuntu.

Is there any instructions on how to build a custom base image that will work with Image Builder?

**Comment by gns-build-server** [comment]
We need to move to Ubuntu 22.04 sooner than later to reap performance benefits over Ubuntu 20.04 without having to go to unsupported third-party repos.  For example, Python 3.10 is tracked by 22.04 and offers 50-60% speedups over Python 3.8 which is the latest tracked by Ubuntu 20.  We are considering working around this issue by adding the deadsnakes repo but the deadsnakes team themselves advise against it in a production setting because they have no mission to keep up with security updates.  Our other alternative would be to build newer versions ourselves and maintain them solo, which is understandably quite messy.

**Comment by pb8o** [comment]
Looks like Ubuntu 22 images are now available:

```
        {
            "arn": "arn:aws:imagebuilder:us-east-1:aws:image/ubuntu-server-22-lts-arm64/x.x.x",
            "name": "Ubuntu Server 22 LTS ARM64",
            "type": "AMI",
            "platform": "Linux",
            "osVersion": "Ubuntu 22",
            "owner": "Amazon",
            "dateCreated": "2023-03-03T03:49:16.708Z"
        },
        {
            "arn": "arn:aws:imagebuilder:us-east-1:aws:image/ubuntu-server-22-lts-x86/x.x.x",
            "name": "Ubuntu Server 22 LTS x86",
            "type": "AMI",
            "platform": "Linux",
            "osVersion": "Ubuntu 22",
            "owner": "Amazon",
            "dateCreated": "2023-03-03T03:49:16.062Z"
        },

```

**Comment by gns-build-server** [comment]
I noticed Ubuntu 22.04 is still not a pulldown option in the Parallelcluster 3.5 GUI.  Not sure if that's expected or not.

**Comment by enrico-usai** [comment]
Hi @gns-build-server, yes this is expected, having Ubuntu 22 support on ImageBuilder is a required condition to enable it in ParallelCluster but it's not enough, we need to adapt ParallelCluster cookbook recipes and CLI code to fully enable it. Please continue to follow https://github.com/aws/aws-parallelcluster/issues/4060 Thanks

**Comment by anderik1** [comment]
This issue should be closed as Ubuntu 22.04 is shown as supported in the docs as well: https://docs.aws.amazon.com/imagebuilder/latest/userguide/what-is-image-builder.html#image-builder-os

**Comment by anderik1** [comment]
This issue should be closed as Ubuntu 22.04 is shown as supported in the docs as well: https://docs.aws.amazon.com/imagebuilder/latest/userguide/what-is-image-builder.html#image-builder-os

**Comment by aaronzielke** [comment]
EC2 Image Builder support for Ubuntu 22.04 managed image is available (since end of March 2023)
